presto-main/src/main/java/io/prestosql/execution/SqlQueryExecution.java

```diff
@@ -519,7 +519,9 @@ public void fail(Throwable cause)
     {
         requireNonNull(cause, "cause is null");
 
-        stateMachine.transitionToFailed(cause);
+        if (stateMachine.transitionToFailed(cause)) {
+            stateMachine.updateQueryInfo(Optional.empty());
```

Comment:
The current method `SqlQueyExecution.fail(Throwable)` is called from many places.  Some of these places are after the query execution has started, so we want to have the final query info.  Putting this call here, will not allow us to wait for the final query info in these cases.  Instead, I think we should move this call to the place that planning is failing.

Reply:
Moved call to point at which planning is failing. Note that the fail() call does not return a boolean to indicate whether the stateMachine transitioned to a failed state, so I'm always calling updateQueryInfo when an exception is caught.